web_requester.go

```diff
@@ -159,11 +159,20 @@ func (w *webRequester) Request() error {
 	// connect to, while the Request's Host field optionally
 	// specifies the Host header value to send in the HTTP
 	// request.
-	if host, ok := w.headers["Host"]; ok {
-		if len(host) != 1 {
-			return errors.New("Host Header is not a single value.")
+
+	var hosts []string
+
+	// headers are case-insensitive
+	for k, v := range w.headers {
```

Comment:
Sorry, this enumeration with ToLower calls is too much of a perf hit.
Let's check `w.headers["Host"]` and `w.headers["host"]` and be done with it.